lib/puppet/type/file/source.rb

```diff
@@ -129,18 +129,6 @@ def checksum
       metadata && metadata.checksum
     end
 
-    # Look up (if necessary) and return local content.
-    def content
-      return @content if @content
-      raise Puppet::DevError, _("No source for content was stored with the metadata") unless metadata.source
-
-      tmp = Puppet::FileServing::Content.indirection.find(metadata.source, :environment => resource.catalog.environment_instance, :links => resource[:links])
-      unless tmp
-        self.fail "Could not find any content at %s" % metadata.source
-      end
-      @content = tmp.content
```

Comment:
It looks like we're losing the ability to cache content. Would there be any value to reading the contents of the file and saving it, then yielding that? I'm not sure if that makes any sense to do now that we're streaming the content though

Reply:
I think this is ok and desirable, otherwise `puppet apply` will keep all file contents in memory for the duration of the run.